src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -387,4 +401,26 @@ TEST_CASE("request_rate_sequence")
   trrm.TestSequences();
 }
 
+TEST_CASE("request_rate_streaming: test that streaming-specific logic works")
+{
+  PerfAnalyzerParameters params{};
+  params.streaming = true;
+
+  TestRequestRateManager trrm(params);
+  trrm.schedule_.push_back(std::chrono::nanoseconds(1));
+
+  std::shared_ptr<TestRequestRateManager::ThreadStat> thread_stat{
+      std::make_shared<TestRequestRateManager::ThreadStat>()};
+  std::shared_ptr<TestRequestRateManager::ThreadConfig> thread_config{
+      std::make_shared<TestRequestRateManager::ThreadConfig>(0, 0)};
+
+  std::future<void> infer_future{std::async(
+      &TestRequestRateManager::Infer, &trrm, thread_stat, thread_config)};
+
+  early_exit = true;
+  infer_future.get();
+
+  CHECK(trrm.stats_->enable_stats == true);
```

Comment:
check that trrm.stats_->num_start_stream_calls == 1
check that trrm.stats_->num_async_stream_infer_calls >= 1

Reply:
That's already tested in `TestInferType()`.

Reply:
https://github.com/triton-inference-server/client/pull/189#discussion_r1013161516

Reply:
I admit, it is a little awkward to add a unit test just for a particular part of an if body. It was the most I could think of when trying to "test streaming in request rate manager".

Reply:
Yes, I see that now. This is worth testing. I see the true case being tested. What I don't see is the false case being tested.  IsDecoupled can be passed into the mock parser just like is_sequence_model is.

Also, I know the variable is "enable_stats" in the StartStream function, but can you rename the variable inside of the stats class to be something else? Otherwise it implies it is an enable flag for that class. Something like start_stream_enable_stats_value

Reply:
Added false case and renamed variable 👍